app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,148 @@
+package com.mealsmadeeasy.ui.home.profile
+
+import android.app.AlertDialog
+import android.app.DatePickerDialog
+import android.content.Context
+import android.os.Bundle
+import android.support.design.widget.TextInputEditText
+import android.support.design.widget.TextInputLayout
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.ui.BaseFragment
+
+import java.util.*
+import android.widget.ArrayAdapter
+import com.mealsmadeeasy.model.Gender
+import com.mealsmadeeasy.model.UserProfile
+import java.text.SimpleDateFormat
+
+
+const val MIN_AGE = 13
+const val INCHES_IN_FEET = 12
+const val MIN_BMI = 15
+const val MAX_BMI = 50
+
+class ProfileFragment : BaseFragment() {
+
+    companion object {
+        fun newInstance(): ProfileFragment = ProfileFragment()
+
+        fun heightToInches(feet: Int, inches: Int): Int {
+            return feet * INCHES_IN_FEET + inches
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val root = inflater.inflate(R.layout.view_user_profile, container, false)
+
+        // Populate sex spinner values
+        val sexSpinner = root.findViewById<Spinner>(R.id.sex_spinner)
+        var sexValues: List<String> = emptyList()
+        for (item in Gender.values()) {
+            sexValues += item.toString()
+        }
+        sexValues += getString(R.string.profile_spinner_prompt)
+        val sexAdapter = HintAdapter(this.activity, android.R.layout.simple_list_item_1, sexValues)
+        sexAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        sexSpinner.adapter = sexAdapter
+        sexSpinner.setSelection(sexAdapter.count)
+
+        // Set up birthday date picker
+        val dateText = root.findViewById<EditText>(R.id.birthday_edit_text)
+        val cal = Calendar.getInstance()
+        setupDatePicker(dateText, cal)
+
+        // Set up text input view hints
+        val feetContainer = root.findViewById<TextInputLayout>(R.id.height_feet_container)
+        val feetEditText = root.findViewById<TextInputEditText>(R.id.height_feet_edit_text)
+        val inchesContainer = root.findViewById<TextInputLayout>(R.id.height_inches_container)
+        val inchesEditText = root.findViewById<TextInputEditText>(R.id.height_inches_edit_text)
+        val weightContainer = root.findViewById<TextInputLayout>(R.id.weight_container)
+        val weightEditText = root.findViewById<TextInputEditText>(R.id.weight_field)
+        feetEditText.onFocusChangeListener = TextInputViewHintListener(feetContainer,
```

Comment:
Can we add a newline before this? I'd like to quarantine the findViewById hell we've got going here.